### PR TITLE
[bug] close write anyway to avoid connection leak

### PR DESCRIPTION
**Pull request by leaves12138**
### Purpose

Close  rewriter anyway no mater what to avoid connection leak.

### CODE REVIEW

**Review by liming30 — APPROVED** [review]
+1

File: paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreWrite.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public AppendOnlyCompactManager.CompactRewriter compactRewriter(
             if (toCompact.isEmpty()) {
                 return Collections.emptyList();
             }
-            RowDataRollingFileWriter rewriter =
+            try (RowDataRollingFileWriter rewriter =
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreWrite.java:150]: Can you use try finally to avoid closing twice?

**Review by JingsongLi — CHANGES_REQUESTED** [review]
See my comment.

File: paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreWrite.java (hunk 1 above)
Comment (leaves12138) [reply paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreWrite.java:150]: Sry forget to remove the close inside.

Comment (liming30) [reply paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreWrite.java:150]: But `rewriter.result()` requires that the `writer` has been closed.

**Review by liming30 — APPROVED** [review]
+1

**Pull request merged**